php/routes.php

```diff
@@ -26,4 +26,42 @@
             ]
         );
     });
+
+    CBPRouter::code('/code', function ($payload) {
+        $parsedUrl = wp_parse_url($payload['url']);
+
+        $isValidHost = $parsedUrl['host'] === 'gist.githubusercontent.com' ||
+            $parsedUrl['host'] === 'raw.githubusercontent.com' ||
+            $parsedUrl['host'] === 'github.com';
+
+        if (!$isValidHost) {
+            return new WP_REST_Response(
+                [
+                    'code' => 'Error: Invalid Host'
+                ]
+            );
+        }
+
+        if ($parsedUrl['host'] === 'github.com') {
+            $parsedUrl['host'] = 'raw.githubusercontent.com';
+            $parsedUrl['path'] = str_replace('/blob', '', $parsedUrl['path']);
+            $payload['url'] = $parsedUrl['scheme'] . '://' . $parsedUrl['host'] . $parsedUrl['path'];
+        }
+
+        $response = wp_remote_get($payload['url']);
+
+        if ($response['headers']['content-type'] !== 'text/plain; charset=utf-8') {
+            return new WP_REST_Response(
+                [
+                    'code' => 'Error: Invalid Content-Type'
+                ]
+            );
+        }
```

Comment:
Do we actually need to check this? If we get back bad data then $response will be of WP_Error type, so we probably just need to do something like 

```php
if (is_wp_error($response)) {
  return new \WP_REST_Response([
    'message' => $response->get_error_message()
  ]);
}
```

Reply:
So this one might actually need some discussion, so in an effort to keep things "safe" (Which may or may not be valid tbh), I figured we should pretty much enforce that links always return text. Teeechnically you can push images/other things to GitHub and you could try and link it and cause unknown behaviors.

I could remove it though and it would give the user some more freedom as far as what they could link